dask/dataframe/groupby.py

```diff
@@ -427,6 +464,9 @@ def _build_agg_args_single(result_column, func, input_column):
     elif func == 'mean':
         return _build_agg_args_mean(result_column, func, input_column)
 
+    elif isinstance(func, Aggregation):
```

Comment:
Generally speaking, the aggregations can be arbitrarily named. Problems only arise, if a function name-clashes for the same column, since then intermediate values will be passed incorrectly and both will use the same output column . 

I added tests to clarify the current behavior and added a descriptive exception for the error case when `.aggregate` is called.